### DISCUSSION TITLE
How can I create an animated GIF with one static background and various foreground 'shining through' PNG?

### DISCUSSION THREAD

**Discussion by PeterCodar** [ImageMagick/ImageMagick#5299] [Help] [answered]
I can't find a working example for this situation:
There are several foreground PNG pictures showing the current rainfall in my area.
I would like to combine several of this rainfall pictures with a static background PNG picture of my country into an animated GIF.

To (nearly) achieve this, I first create an "AnimatedRain.gif" with all my rainfall pictures combined with this command:
`convert -delay 20 -dispose previous rainfall*.png[150%] -loop 0 AnimatedRain.gif`
(using [150%] to enlarge the single rainfall pictures matching the background country picture in the next step)

Then I try to combine this AnimatedRain.gif with my static background PNG with:
`convert StaticMap.gif ( AnimatedRain.gif -repage 0x0+0+0 ) -loop 0 AnimatedRain_withstatic_Background.gif`

What I can't get working:
How can I create such an animated GIF with 'shining through / transparent' rainfall pictures?
The idea is that I still can read the underlaying city names when the rain passes over them

![AnimatedRain](https://user-images.githubusercontent.com/18629146/177313146-dfd1ddc1-6a92-42d8-85fc-cf23da0a7cfb.gif)

IM: ImageMagick-7.0.11-2-portable-Q16-x64

**Comment by fmw42** [comment]
Try:

`magick StaticMap.gif null: ( AnimatedRain.gif -coalesce ) -compose over -layers composite -layers optimize -loop 0 AnimatedRain_withstatic_Background.gif
`

Note for Imagemagick 7, use magick, not convert.

See https://imagemagick.org/Usage/anim_mods/#composite

**Comment by PeterCodar** [comment]
Thank you fmw42

Unfortunately, this doesn't make the animated rain "shining through". It combines the animated rain with the static background gif.
But city names with heavy rainfall (yellow and red) can't be read with the above `magick` command.

**Reply by fmw42** [reply]
Use a different compose method. Or post your images so that we can work with them

**Reply by PeterCodar** [reply]
In the meantime, I have tried all 14 possible compose methods to no avail.
Sorry if I'm annoying you, but could you please try it with the `AnimatedRain.gif` from my first post and the attached [StaticMap.gif](https://user-images.githubusercontent.com/18629146/177578926-522ccf29-a8eb-4e96-809b-a3b9e28b944b.gif) and tell me your working method? I'm looking for commands, where the static background city like _Chambéry_ remain visible even if the rain passes over.

**Comment by fmw42** [comment, marked as answer]
Is this what you want?

`magick StaticMap.gif null: ( AnimatedRain.gif -coalesce ) -define compose:args=60,100 -compose dissolve -layers composite -layers optimize -loop 0 AnimatedRain_withstatic_Background.gif`

![AnimatedRain_withstatic_Background](https://user-images.githubusercontent.com/13458218/177588244-9af2a427-c041-4c36-b12a-022ea50d6e62.gif)

**Reply by PeterCodar** [reply]
This looks great! I try this immediately, just a moment...

WOW, I'm really impressed, I would never have come up with this solution. 
Thank you very much @fmw42! Highly appreciated.
